rfc/src/rfcs/0008-source-coverage.md

```diff
@@ -0,0 +1,277 @@
+by default. For example, the [draft implementation](https://github.com/model-checking/kani/pull/3119)
+writes the coverage files into the `target/kani/<target_triple>/cov/` directory.
+
+Users aren't expected to read the information in any of these files.
+Therefore, there's no need to restrict their format.
+The [draft implementation](https://github.com/model-checking/kani/pull/3119)
+uses the JSON format but we might consider using a binary format if it doesn't
+scale.
+
+[^note-kanimap]: Note that the `.kanimap` generation isn't implemented in [#3119](https://github.com/model-checking/kani/pull/3119).
+The [draft implementation of `kani-cov`](https://github.com/model-checking/kani/pull/3121)
+simply reads the source files referred to by the code coverage checks, but it
+doesn't get information about code trimmed out by the MIR linker.
+
+#### The `kani-cov` tool
+
+The `kani-cov` tool will be used to process coverage information generated by
+Kani and produce coverage outputs as indicated by the user.
+Hence, the `kani-cov` tool corresponds to the set of LLVM tools
+(`llvm-profdata`, `llvm-cov`, etc.) that are used to produce coverage outputs
+through the LLVM coverage workflow.
+
+In contrast to LLVM, we'll have a single tool for all Kani coverage-related needs.
+We suggest that the tool initially offers three subcommands[^note-export]:
+ - `merge`: to combine the coverage results of one or more `.kaniraw` files into
+ a single `.kanicov` file, which will be required for the other subcommands.
+ - `report`: to display a summary of the coverage results.
+ - `show`: to produce source-based code coverage reports in human-readable formats (e.g., HTML).
+
+
+Let's assume that we've run `cargo kani cov` and generated coverage files in the `my-coverage` folder.
+Then, we'd use `kani-cov` as follows to combine the coverage results[^note-exclude] for all harnesses:
+
+```sh
+kani-cov merge my-coverage/*.kaniraw -o my-coverage.kanicov
+```
+
+Let's say the user is first interested in reading a coverage summary through the terminal.
+They will have to use the `report` command for that:
+
+```sh
+kani-cov report my-coverage/default.kanimap -instr-profile=my-coverage.kanicov --show-summary
+```
+
+The command could print a coverage summary like:
+
+```
+| Filename | Regions | Missed Regions | Cover | Functions | ...
+| -------- | ------- | -------------- | ----- | --------- | ...
+| main.rs  |       9 |              3 | 66.67 |         3 | ...
+[...]
+```
+
+Now, let's say the user wants to produce an HTML report of the coverage results.
+They will have to use the `show` command for that:
+
+```sh
+kani-cov show my-coverage/default.kanimap -format=html -instr-profile=my-coverage.kanicov -o coverage-report
+```
+
+This time, the command will generate a `coverage-report` folder including a
+browsable HTML webpage that highlights the regions covered in the source
+according to the coverage results in `my-coverage.kanicov`.
+
```

Comment:
What about IDE integration? (However, a quick search seems to indicate that the [VSCode code coverage plugin](https://github.com/markis/vscode-code-coverage) only highlights whole lines. That would be a good reason to only support terminal and HTML. But I still think it's worth calling out explicitly)

Reply:
Sorry, I've pushed a new subsection to discuss the integration with the Kani VS Code Extension. To integrate with that particular plugin we'd just need to be able to export to the LCOV format. Let me call it out there.